lib/ansible/modules/crypto/openssh_keypair.py

```diff
@@ -0,0 +1,355 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2018, David Kainz <dkainz@mgit.at> <dave.jokain@gmx.at>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: openssh_keypair
+author: "David Kainz (@lolcube) aka (@bsod)"
+version_added: "2.8"
+short_description: Generate OpenSSH private and public keys.
+description:
+    - "This module allows one to (re)generate OpenSSH private and public keys. It uses
+      ssh-keygen to generate keys. One can generate rsa, dsa, rsa1, ed25519
+      or ecdsa private keys."
+requirements:
+    - "ssh-keygen"
+options:
+    state:
+        required: false
+        default: present
+        choices: [ present, absent ]
+        description:
+            - Whether the private and public keys should exist or not, taking action if the state is different from what is stated.
+    size:
+        required: false
+        description:
+            - "Specifies the number of bits in the private key to create. For RSA keys, the minimum size is 1024 bits and the default is 4096 bits.
+              Generally, 2048 bits is considered sufficient.  DSA keys must be exactly 1024 bits as specified by FIPS 186-2.
+              For ECDSA keys, size determines the key length by selecting from one of three elliptic curve sizes: 256, 384 or 521 bits.
+              Attempting to use bit lengths other than these three values for ECDSA keys will cause this module to fail.
+              Ed25519 keys have a fixed length and the size will be ignored."
+    type:
+        required: false
+        default: rsa
+        choices: ['rsa', 'dsa', 'rsa1', 'ecdsa', 'ed25519']
+        description:
+            - The algorithm used to generate the SSH private key.
+    force:
+        required: false
+        default: false
+        type: bool
+        description:
+            - Should the key be regenerated even if it already exists
+    path:
+        required: true
+        description:
+            - Name of the file in which the generated SSH private and public key will be written.
+    passphrase:
+        required: false
+        description:
+            - The passphrase for the private key.
+    comment:
+        required: false
+        description:
+            - Provides a new comment to the public key
```

Comment:
You should note that the comment is not used when checking whether an existing key matches the module's options (i.e. for idempotency).